### PR TITLE
#6871 fixing the visibility of map detail panel which toggling 3D

### PR DESCRIPTION
**Pull request by QSafariallahkheili**
## Description

It solves the conflict between tutorial and detail panel
**Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [x] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#6871

**What is the new behavior?**

It solves the conflict between tutorial and detail panel when toggling on 3D
## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

File: web/client/actions/details.js [hunk 1]
```diff
@@ -39,9 +39,9 @@ export const detailsLoaded = (mapId, detailsUri, detailsSettings) => ({
  * @memberof actions.details
  * @return {action}        type `OPEN_DETAILS_PANEL`
 */
-export const openDetailsPanel = () => ({
-    type: OPEN_DETAILS_PANEL
-});
+export const openDetailsPanel = () => {
```
Comment (MV88) [inline web/client/actions/details.js:42]: can you revert this change since its not needed?

File: web/client/epics/globeswitcher.js [hunk 2]
```diff
@@ -6,26 +6,27 @@
...
  * @param {external:Observable} action$ manages `TOGGLE_3D`.
  * @return {external:Observable} emitting connected-react-router push action and {@link #actions.globeswitcher.updateLast2dMapType} actions
  */
 export const updateRouteOn3dSwitch = (action$, store) =>
+
     action$.ofType(TOGGLE_3D)
         .switchMap((action) => {
             const last2dMapType = last2dMapTypeSelector(store.getState());
-            return Rx.Observable.of(changeMapType(action.originalMapType !== "cesium" ? "cesium" : last2dMapType));
+            // two actions are called after toggling to 3D
```
Comment (MV88) [inline web/client/epics/globeswitcher.js:24]: this comment is superflous, please remove it
```suggestion
            // two actions are called after toggling to 3D
```

File: web/client/epics/tutorial.js [hunk 3]
```diff
@@ -159,10 +161,19 @@ export const getActionsFromStepEpic = (action$) =>
  * @type {Object}
  */
 
+export const openDetailsPanelEpic = (action$, store) =>
```
Comment (MV88) [inline web/client/epics/tutorial.js:164]: can you try to add a unit test for this?

**Review by MV88 — CHANGES_REQUESTED** [review]
there is a problem also when you open directly the cesium viewer and not only when passing to 3D from 2D

if you open directly this url the problem of the conflict is still present
http://localhost:8081/?debug=true#/viewer/cesium/34133

@QSafariallahkheili 
in the epic **storeDetailsInfoEpic** when we choose to dispatch the openDetailsPanel we first need to check if a tutorial is in progress by checking its state
some something like

```js
const isTutorialRunning = store.getState().tutorial.run;
return Rx.Observable.of(
                            detailsLoaded(mapId, details.value, detailsSettings),
                            ...(detailsSettings.showAtStartup && !isTutorialRunning ? [openDetailsPanel()] : [])
                        );
``

please try to replicate what i'm reporting and then try the suggested fix

**Review by MV88 — APPROVED** [review]

**Review by MV88 — CHANGES_REQUESTED** [review]
there are tests that are failing,
can you check them locally?

you can run  first time "npm test"

and see which tests are failing and why

**Review by MV88 — APPROVED** [review]

**Comment by ElenaGallo** [comment]
@QSafariallahkheili a note after testing this issue:

If the tutorial is opened from the Options menu (it does not open at startup), the Details tool will also open

**How to rerpoduce**
- Open 
- Click on 3D 
- Close the Tutorial
- Click on Options menu
- Click on Tutorial
- Click Next to the end of the tutorial


*Current Reuslt*
Opening the tutorial from the Options menu should not open the details tool as well

**Pull request merged**